src/dynimage.rs

```diff
@@ -133,7 +133,14 @@ impl DynamicImage {
     pub fn from_decoder<'a>(decoder: impl ImageDecoder<'a>)
         -> ImageResult<Self>
     {
-        decoder_to_image(decoder)
+        decoder_to_image(decoder, DecodingLimits::default())
+    }
+
+    /// Decodes an encoded image into a dynamic image.
+    pub(crate) fn from_decoder_limits<'a>(decoder: impl ImageDecoder<'a>, limits: DecodingLimits)
```

Comment:
What is the point of this method if it just directly calls decoder_to_image? Is there some issue with the generics?

Reply:
I just copied the `from_decoder` method, but you are right. This would be simpler with just calling `decoder_to_image`

Reply:
Doesn't this duplicate implementations? It would introduce an array of different `set_*` methods that will need to be added to both `Reader` and `ImageDecoder`. Would grouping those parameters into the `Limits` struct not be equivalent but require less interfaces? Chances are that the decoders will store many of the same resource restrictions internally.